File: exam-slammr-lambda/exam-slammr-lambda-question/src/main/java/uk/co/claritysoftware/exam/slammr/lambda/question/dynamodb/items/AnswerDocument.java
package uk.co.claritysoftware.exam.slammr.lambda.question.dynamodb.items;

import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAttribute;
import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBDocument;

import lombok.Builder;
import lombok.Value;

/**
 * @author Nathan Russell
 */
@Value
@Builder
@DynamoDBDocument
public class AnswerDocument {

	@DynamoDBAttribute
	private final String text;

	@DynamoDBAttribute
	private final boolean correct;

}

File: exam-slammr-webapp/src/main/kotlin/uk/co/claritysoftware/exam/slammr/webapp/persistence/dynamodb/item/question/AnswerDocument.kt
package uk.co.claritysoftware.exam.slammr.webapp.persistence.dynamodb.item.question

import com.amazonaws.annotation.Immutable
import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBDocument
import java.io.Serializable

/**
 * Encapsulates the data for a potential Answer to a {@link QuestionItem} dynamodb item
 */
@DynamoDBDocument
@Immutable
data class AnswerDocument(var text: String, var correct: Boolean) : Serializable